src/Numerics/Optimizers/ExhaustiveOptimizer/exhaustiveoptimizer.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+#=========================================================================
+#
+#  Copyright NumFOCUS
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#
+
+# Python example demonstrating itk.ExhaustiveOptimizer usage
+import sys
+from math import pi
+
+import itk
+
+EXIT_SUCCESS = 0
+EXIT_FAILURE = 1
+
+def main(argv):
+    if(len(argv) < 3):
+        raise Exception(f'Usage: {argv[0]} fixedImage movingImage')
+
+    dimension = 2
+    FixedImageType = itk.Image[itk.F, dimension]
+    MovingImageType = itk.Image[itk.F, dimension]
+    TransformType = itk.Euler2DTransform[itk.D]
+    OptimizerType = itk.ExhaustiveOptimizerv4[itk.D]
+    MetricType = itk.MeanSquaresImageToImageMetricv4[FixedImageType, MovingImageType]
+    TransformInitializerType = \
+        itk.CenteredTransformInitializer[itk.MatrixOffsetTransformBase[itk.D,2,2],
+                                         FixedImageType, MovingImageType]
+    RegistrationType = itk.ImageRegistrationMethodv4[FixedImageType,MovingImageType]
+
+    fixedImage = itk.imread(argv[1], itk.F)
+    movingImage = itk.imread(argv[2],itk.F)
+    transform = TransformType.New()
+    metric = MetricType.New()
```

Comment:
Since the `MetricType` declaration is long, it can be two lines. However, keep the type declaration next to the instantiations for readability.

Reply:
For classes such as `TransformInitializerType` and `RegistrationType` not instantiated right away, is it better practice to define types at the top of the function or next to the instantiations?

Reply:
The type declarations should be next to the instantiations -- this is the case for the C++ examples, too. This improved scoping and locality improves readability.